docs/topics/class-based-views/mixins.txt

```diff
@@ -686,3 +686,123 @@ Because of the way that Python resolves method overloading, the call to
 ``super().render_to_response(context)`` ends up calling the
 :meth:`~django.views.generic.base.TemplateResponseMixin.render_to_response()`
 implementation of :class:`~django.views.generic.base.TemplateResponseMixin`.
+
+.. _view-init-example:
+
+Mixins setting view attributes
+==============================
+
+Consider an application allowing administrators to impersonate users and act on
+their behalf, for example recording interest for an author. The impersonation
+logic is a bit tricky and has been encapsulated in a mixin to facilitate reuse.
+
+.. code-block:: python
+    :caption: views.py
+
+    from django.contrib.auth.models import User
+    from django.http import Http404, HttpResponse
+    from django.shortcuts import get_object_or_404
+    from django.views import View
+    from django.views.generic.detail import SingleObjectMixin
+
+    from books.models import Author
+
+
+    class ImpersonationMixin:
+        def dispatch(self, request, *args, **kwargs):
+            if 'username' in kwargs and request.user.is_staff:
+                self.user = get_object_or_404(User, username=kwargs['username'])
+            elif 'username' in kwargs or not request.user.is_authenticated:
+                # Anonymous and regular users cannot impersonate others.
+                raise Http404
+            else:
+                self.user = request.user
+            return super().dispatch(request, *args, **kwargs)
+
+
+    class RecordInterest(ImpersonationMixin, SingleObjectMixin, View):
+        model = Author
+
+        def post(self, request, *args, **kwargs):
+            author = self.get_object()
+            author.record_interest_from(self.user)
+            return HttpResponse('Interest saved successfully.')
+
+.. code-block:: python
+    :caption: urls.py
+
+    from django.urls import path
+    from books.views import RecordInterest
+
+    urlpatterns = [
+        #...
+        path('author/<int:pk>/interest/', RecordInterest.as_view()),
+        path('impersonate/<username>/author/<int:pk>/interest/', RecordInterest.as_view()),
+    ]
+
+Now, let's prevent administrators from impersonating inactive users, returning
+a :class:`~django.http.HttpResponseForbidden` when an administrator
+impersonates inactive users. A naive approach could be to extend
+``ImpersonationMixin`` as follows:
+
+.. code-block:: python
+    :caption: views.py
+
+    class ActiveUserImpersonationMixin(ImpersonationMixin):
+        def dispatch(self, request, *args, **kwargs):
+            # CAUTION: Raises AttributeError, self.user is not (yet) set.
+            if not self.user.is_active:
+                return HttpResponseForbidden('User is inactive.')
+            # Generates a HttpResponse, must be called after user status check.
+            return super().dispatch(request, *args, **kwargs)
```

Comment:
Hmmm. I still have real concerns about showing users that they should move core view logic into dispatch... It might work in a particular shop, but I worry we're just teaching people to put their logic all over the place, rather than straight-line in front of them. 

(Just an observation.)